gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+      }
+    }
+
+    private String toJson(String message) throws IOException {
+      Map<String, Object> args = new HashMap<>();
+      Map<String, Object> errors = new HashMap<>();
+      errors.put("domain", domain);
+      errors.put("message", message);
+      errors.put("reason", reason);
+      args.put("errors", ImmutableList.of(errors));
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      return jsonFactory.toString(ImmutableMap.of("error", args));
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      // see https://cloud.google.com/resource-manager/reference/rest/
+      String path = exchange.getRequestURI().getPath();
+      String requestMethod = exchange.getRequestMethod();
+      Response response = null;
+      if (requestMethod.equals("POST") && path.startsWith("/" + VERSION + "/projects")) {
+        if (path.contains("undelete")) {
+          try {
+            response = undelete(projectIdFromURI(path));
+          } catch (IOException e) {
+            response = Error.BAD_REQUEST.response(e.getMessage());
+          }
+        } else {
+          String requestBody =
+              decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+          response = create(jsonFactory.fromString(requestBody, Project.class));
+        }
+      } else if (requestMethod.equals("DELETE")) {
+        response = delete(projectIdFromURI(path));
+      } else if (requestMethod.equals("GET")) {
+        if (path.startsWith("/" + VERSION + "/projects/")) {
+          response = get(projectIdFromURI(path), parseFields(exchange.getRequestURI().getQuery()));
+        } else {
+          response = list(parseListOptions(exchange.getRequestURI().getQuery()));
+        }
+      } else if (requestMethod.equals("PUT")) {
+        String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+        response = replace(jsonFactory.fromString(requestBody, Project.class));
+      }
+      response = firstNonNull(
+          response, Error.BAD_REQUEST.response("The server could not understand the request."));
+      exchange.getResponseHeaders().set("Content-type", "application/json; charset=UTF-8");
+      exchange.sendResponseHeaders(response.code(), response.body().length());
+      OutputStream outputStream = exchange.getResponseBody();
+      outputStream.write(response.body().getBytes(StandardCharsets.UTF_8));
+      outputStream.close();
+    }
+  }
+
+  private static String decodeContent(Headers headers, InputStream inputStream) throws IOException {
+    List<String> contentEncoding = headers.get("Content-encoding");
+    InputStream input = inputStream;
+    if (contentEncoding != null && contentEncoding.size() > 0
+        && contentEncoding.get(0).contains("gzip")) {
```

Comment:
Nit, should be "gzip" or ""x-gzip". see [here](http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.5). You don't to try it on "bzip". Any other compression should be rejected with "Unsupported".